src/OpenTelemetry.AutoInstrumentation/AutoInstrumentationEventSource.cs

```diff
@@ -0,0 +1,46 @@
+// <copyright file="AutoInstrumentationEventSource.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System.Diagnostics.Tracing;
+
+namespace OpenTelemetry.AutoInstrumentation;
+
+/// <summary>
+/// EventSource implementation for OpenTelemetry AutoInstrumentation implementation.
+/// </summary>
+[EventSource(Name = "OpenTelemetry-AutoInstrumentation")]
+internal class AutoInstrumentationEventSource : EventSource
+{
+#pragma warning disable SA1401 // Fields should be private
+    public static AutoInstrumentationEventSource Log = new AutoInstrumentationEventSource();
```

Comment:
Consider Lazy field/property.

Reply:
Putting together with the private constructor the static field becomes the only way to access the instance methods. In light of that it is not necessary to lazy initialize it: the runtime takes care of that (if not used never initialized).

Add `readonly` too.

Reply:
This is for EventSource data collection, we won't be able to mark this as lazy. This is being utilized by framework itself and not initialized in auto-instrumentation.

Reply:
Reference from OpenTelemetry SDK - https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/src/OpenTelemetry/Internal/OpenTelemetrySdkEventSource.cs

Reply:
@rajkumar-rangaraj `readonly` seems a good addition to the static field (I think you may have missed my comment because I published it around the same time as your comments above).

Reply:
Changed to `readonly` - 02500098e5714212260ab4c78e89942bcc509842